app/components/header/header.component.html

```diff
@@ -12,19 +12,29 @@
       </li>
       <li class="hidden-xs"><img src="./assets/RightArrow.svg" class="right-arrow"></li>
       <li class="branch-name dropdown hidden-xs">
-        <a href="#" class="dropdown-toggle" id="branch-name" data-toggle="dropdown" onclick="switchToMainPanel()">
+        <a href="#" class="dropdown-toggle" id="branch-name" data-toggle="dropdown" onclick="switchToMainPanel();displayBranchesTags()">
           branch<span class="caret"></span>
         </a>
         <ul class="dropdown-menu" id="branch-dropdown" role="menu" aria-labelledby="branch-name">
-          <li role="presentation" id="create-branch">
-            <div class="input-group menuitem">
-              <input type="text" id="branchName" class="form-control" placeholder="Search or create branch">
-              <span class="input-group-btn">
-                <button class="btn btn-default" type="button" onclick="createBranch()">OK</button>
-              </span>
-            </div>
-          </li>
-        </ul>
+            <li role="presentation" id="create-branch">
+              <div class="select-menu-header">
+                <span class="select-menu-title"> Switch branches/tags</span>
+              </div>
+              <div class="input-group menuitem">
+                <div class="select-menu-text-filter" style="padding:10px">
+                    <span class="input-group-btn">
+                    <input type="text" id="branchName" class="form-control" style="width:320px" placeholder="Find or create branch" oninput="displayBranchesTags()">
+                    <button class="btn btn-default" type="button" onclick="createBranch()">OK</button>
+                    </span>
+                </div>
+                <button class="btn btn-default" style="width:45%; margin: 0 1% 5px 4%" (click)=displayBranches()>Branches</button>
```

Comment:
The CSS style should be implemented from a linked component style sheet (i.e. header.component.css) or inline in header.component.ts